client/src/components/User/UserBeaconSettings.vue

```diff
@@ -0,0 +1,259 @@
+          </div>
+
+          <!-- Case: History exists -->
+          <div v-if="beaconHistory.id" class="flex-row">
+            <div class="no-shrink">
+              Variant files are searched in your beacon history
+            </div>
+            <div class="fill"></div>
+            <div class="no-shrink">
+              <b-button @click="switchHistory(beaconHistory.id)">Switch to History</b-button>
+            </div>
+          </div>
+        </div>
+
+        <div v-if="enabled">
+          <p>
+            Datasets must fulfill the following conditions in order to be processed
+          </p>
+          <ul>
+            <li>must be VCF or VCF.bzip format</li>
+            <li>must have a human reference assigned to it (e.g. hg19)</li>
+            <li>must define at least one sample</li>
+          </ul>
+        </div>
+      </b-modal>
+    </div>
+  </b-row>
+</template>
+
+<script>
+import store from "store"
+import axios from "axios";
+import Vue from "vue";
+import BootstrapVue from "bootstrap-vue";
+
+Vue.use(BootstrapVue);
+
+export default {
+  props: {
+    root: {
+      type: String,
+      required: true,
+    },
+    userId: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      enabled: false,
+      beaconHistoryName: "___BEACON_PICKUP___",
+      beaconHistory: {}
+    };
+  },
+  created() {
+    this.getBeaconHistory();
+  },
+  methods: {
+    switchHistory: async function (historyId) {
+      await store.dispatch("history/setCurrentHistory", historyId);
+    },
+    getBeaconHistory: function () {
+      axios
+          .get(this.root + "api/histories?q=name&qv=" + this.beaconHistoryName)
+          .then((response) => {
+            if (response.data.length > 0) {
+              // always select first match
+              this.beaconHistory = response.data[0];
+
+            }
+          })
+          .catch((error) => {
+            console.log(error.response)
+          });
+    },
+    createBeaconHistory() {
+      const annotation = "Variant files will be collected from this history if beacon sharing is activated"
+      axios
+          .post(this.root + "api/histories", {name: this.beaconHistoryName})
+          .then((response) => {
+            this.beaconHistory = response.data
+            axios.put(`${this.root}api/histories/${this.beaconHistory.id}`, {"annotation": annotation});
+          })
+          .catch((error) => {
+            this.errorMessages.push(error.response.data.err_msg), console.log(error.response)
+          });
+    },
+    optIn() {
+      try {
+        axios.post(`${this.root}api/users/${this.userId}/beacon`, {"enabled": true}).then(
+            response => {
+              // TODO check response
+              this.loadSettings();
+              if (!this.beaconHistory.id) {
+                this.createBeaconHistory()
+              }
+            }
+        );
+      } catch (e) {
+        console.log(e);
+      }
+    },
```

Comment:
I'm wondering: currently the special Beacon history gets discovered by name. While that just might work, I can see users starting to experiment with creating duplicates of that history potentially under the same name, etc.
Would it not be possible, instead of recording a boolean to store here the actual ID of the Beacon history?

Reply:
Yes, users can absolutely create duplicates which will then also act as beacon histories.
But I see it as a win, that the mechanism is somewhat transparent to the user.

It should be possible to store the history ID in the setting and use that. But the user will still experiment, even more so since now even the name can be changed without changing the behavior.
With a new name nothing "visual" about the history would be left that identifies it as the beacon history.

Therefore I would rather identify histories by a special name. 

We could however combine the two - discover histories by name but then only use the one that has its ID stored in the users beacon setting.

This would eliminate duplicate histories, but has its own caveats. For example when we "disable" the setting by clearing the history ID from the preference, we could get problems when re-activating it and trying to discover the previous history (the duplicates might be back :laughing: )

Other possibilities are to embrace the duplicate histories and display them all in the beacon settings or just leave everything as it is now

Reply:
As discussed, I [changed the modal](https://github.com/usegalaxy-eu/galaxy/pull/131/commits/fabf48ec7ce13fa96ac0894aa00efd32b51c072d) to display all histories that will be used for sharing
![histories](https://user-images.githubusercontent.com/15705080/200273773-131ef3ea-3b8e-44e5-8fd5-01b21bc76944.gif)